src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -124,6 +124,12 @@ internal partial class DiagnosticIncrementalAnalyzer
                     };
                 }
             }
+
+            static bool IsVisibleDocumentOrOpenDocumentWithPriorReportedVisibleDiagnostics(
+                bool isVisibleDocument,
+                bool isOpenDocument,
+                DocumentAnalysisData previousData)
+                => isVisibleDocument || (isOpenDocument && previousData.Items.Any(d => d.Severity is DiagnosticSeverity.Error or DiagnosticSeverity.Warning or DiagnosticSeverity.Info));
```

Comment:
```suggestion
                => isVisibleDocument || (isOpenDocument && previousData.Items.Any(static d => d.Severity is DiagnosticSeverity.Error or DiagnosticSeverity.Warning or DiagnosticSeverity.Info));
```

Reply:
also, i'd just make this a method with normal statements :)

Reply:
Done, thanks